clients/drcachesim/tools/func_view.cpp

```diff
@@ -229,13 +241,15 @@ func_view_t::process_memref(const memref_t &memref)
         bool was_nested = shard->nesting_level > 0;
         if (shard->prev_noret)
             --shard->nesting_level;
-        std::string indent(shard->nesting_level * 4, ' ');
         // Print a "Tnnn" prefix so threads can be distinguished.
         std::cerr << ((was_nested && shard->prev_was_arg) ? "\n" : "") << "T" << std::dec
                   << std::left << std::setw(8) << memref.marker.tid
                   << std::right /*restore*/;
-        std::cerr << indent << "0x" << std::hex << memref.marker.marker_value << " => "
-                  << *id2info_[shard->last_func_id].names.begin() << "(";
+        const std::string name = info.names.empty()
+            ? "<F#" + std::to_string(shard->last_func_id) + ">"
```

Comment:
I see that syscalls exit above, so this can't be a syscall: in which case it's an id missing from the funclist file?  It seems better to either A) make it a fatal error to find a non-syscall id that's not in the function list; B) don't require a funclist at all (today it's an error up front to not provide one): i.e., make it optional, and then printing the number here makes sense.  Requiring the funclist but then not complaining on an invalid id doesn't seem right, esp w/ no comment indicating that this is something unusual: the code looks like this is something normal.

Reply:
You know much more about how normal or reasonable a missing function name is than I do.  The description at the head of this file just says "it optionally uses modules.log file to qualify function names for offline traces" which suggests the names are optional, and furthermore says nothing about whether or not any provided name list has to be complete.  And the application where I discovered this problem actually bypasses the function name file and provides names from another source, where again the optionality and completeness requirements for the name list are unspecified.

So given the _stated_ requirements, I think using F#id as the name for otherwise unnamed functions is useful and provides clear enough indication to the user that any provided funclist didn't have a name for this function.  But I'd be fine taking a tougher line on this if we want to define and enforce stricter name requirements.

Reply:
Per offline discussion I think we arrived at a decision to have the funclist be optional, but when it is supplied it has to match the trace exactly (to catch user errors in mixing up which file goes with which traces) and it's an error if it doesn't.  Was your plan to do that in this PR or that would be something separate?

Reply:
Per additional offline discussion, we decided that since funclist is somewhat integral to how func_view currently operates, it's better to make it mandatory.  So I updated func_view to detect missing or inconsistent funclist info and report errors as appropriate.